### ISSUE TITLE
bugprone-branch-clone false positive in if else if chains with other branches inbetween

### ISSUE THREAD

**Issue by lukas-lang** [llvm/llvm-project#96298]
The following warns about duplicated branches (https://godbolt.org/z/M4WMxxqfs):

```cpp
int test2(int);

void test1(bool a, bool b, bool c) {
    if (a)
        test2(1);
    else if (b)
        test2(0);
    else if (c)
        test2(1);
}
```

Of course, I could rewrite it as follows, but this repeats the conditions, so it's not really better. On the contrary, this will become worse if more branches are in between the repeated ones, and negating the conditions properly can also be confusing to get right:

```cpp
int test2(int);

void test1(bool a, bool b, bool c) {
    if (a || (!b && c))
        test2(1);
    else if (b)
        test2(0);
}
```
